Replace server author services specific data generators with fake-data-generator

In the utils server we are using too many custom functions for the authors fake data. We could replace that using the [fake-data-generator](https://github.com/Cambalab/fake-data-generator) as a dev-dependency.

Current models: please respect the current type of each property

```javascript
author: {
  id: Number, // faker random number
  name: String, // faker random words
  lastname: String, // faker lorem text
  birthdate: ISOString // faker date between
}
```

Checkpoints:

+   [x] ***Run the e2e tests, they should pass.***
+   [x] ***Run the unit tests, they should pass.***
+   [x] **Authors service:** Replace ipsum dependency and the custom date format functions with fake-data-generator. Replace the **authors** array with a new fake-data-generator model of **40 items**
+   [x] ***Rerun e2e tests, they must pass.***
+   [x] ***Rerun unit tests, they must pass.***

Hi !

I tried to do some test with the fake-data-generator but it does not seems to work as expected.

I tried the following code : 

```js
// Requires the package
const { generateModel } = require('fake-data-generator')
// Requires a model
const model = require('./models/authors.json')
// Generate the model
const amountArg = 50
const modelArg = 'authors'
const outputType = 'object'
const generatedModel = generateModel({ amountArg, modelArg, outputType })
```

if `authors` does not exist in the `node_modules/fake-data-generator/dist/models/`, I have the following error : 

```
 jeremy@ArchLinux  ~/Document/Code/test-fake-data  node index.js                                                                                              SIGINT(2) ↵  10406  14:21:32
internal/modules/cjs/loader.js:797
    throw err;
    ^

Error: Cannot find module '../models/authors'
Require stack:
- /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js
- /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/index.js
- /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/main.js
- /home/jeremy/Document/Code/test-fake-data/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:794:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:849:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at getModel (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:17:19)
    at createSingle (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:25:15)
    at /home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:34:12
    at Array.map (<anonymous>)
    at createMany (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:33:47)
    at exports.default (/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js:49:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/create-model.js',
    '/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/lib/index.js',
    '/home/jeremy/Document/Code/test-fake-data/node_modules/fake-data-generator/dist/main.js',
    '/home/jeremy/Document/Code/test-fake-data/index.js'
  ]
}
```

I looked at the `fake-data-generator` code and this [line](https://github.com/Cambalab/fake-data-generator/blob/master/lib/create-model.js#L4) seems to be the problem

So here the question,

Do I need to make a pull request to `fake-data-generator` to add the `authors.json` model or maybe replace 

```js
const modelData = require(`../models/${modelName}`)
```

by 

```js
const modelData = require(path.resolve(`${__dirname}/../models/${modelName}`))
```

to get the current path ?

--- a/utils/server-test/package.json
+++ b/utils/server-test/package.json
@@ -13,6 +13,7 @@
     "body-parser": "^1.18.2",
     "cors": "^2.8.4",
     "express": "^4.16.2",
+    "fake-data-generator": "^0.1.10",
     "jsonwebtoken": "^8.5.0"
   },
   "author": "",

--- a/utils/server-test/server.js
+++ b/utils/server-test/server.js
@@ -1,33 +1,33 @@
 /* eslint-disable import/no-extraneous-dependencies */
-const express = require("express");
-const bodyParser = require("body-parser");
+const express = require('express')
+const bodyParser = require('body-parser')
 
-const createMagazinesService = require("./services/magazines");
-const createArticlesService = require('./services/articles');
-const createAuthorsService = require('./services/authors');
-const createAuthService= require('./services/auth');
+const createMagazinesService = require('./services/magazines')
+const createArticlesService = require('./services/articles')
+const createAuthorsService = require('./services/authors')
+const createAuthService = require('./services/auth')
 
 /* eslint-enable */
 
-var cors = require("cors");
-const app = express();
+var cors = require('cors')
+const app = express()
 
-app.use(cors());
-app.options("*", cors());
+app.use(cors())
+app.options('*', cors())
 
-app.use(express.static(__dirname));
+app.use(express.static(__dirname))
 
-app.use(bodyParser.json());
+app.use(bodyParser.json())
 
-createArticlesService(app);
-createMagazinesService(app);
-createAuthService(app);
-createAuthorsService(app);
+createArticlesService(app)
+createMagazinesService(app)
+createAuthService(app)
+createAuthorsService(app)
 
-const port = process.env.PORT || 8080;
+const port = process.env.PORT || 8080
 
 module.exports = app.listen(port, () => {
   /* eslint-disable no-console */
-  console.log(`Server listening on http://localhost:${port}, Ctrl+C to stop`);
+  console.log(`Server listening on http://localhost:${port}, Ctrl+C to stop`)
   /* eslint-enable no-console */
-});
+})

--- a/utils/server-test/services/articles.js
+++ b/utils/server-test/services/articles.js
@@ -1,92 +1,91 @@
-const Ipsum = require("bavaria-ipsum");
+const Ipsum = require('bavaria-ipsum')
 
-module.exports = function (app) {
-
-  const ipsum = new Ipsum();
+module.exports = function(app) {
+  const ipsum = new Ipsum()
 
   const articles = [
     {
       id: 1,
       title: ipsum.generateSentence(),
-      content: ipsum.generateParagraph()
+      content: ipsum.generateParagraph(),
     },
     {
       id: 2,
       title: ipsum.generateSentence(),
-      content: ipsum.generateParagraph()
+      content: ipsum.generateParagraph(),
     },
     {
       id: 3,
       title: ipsum.generateSentence(),
-      content: ipsum.generateParagraph()
-    }
-  ];
+      content: ipsum.generateParagraph(),
+    },
+  ]
 
-  app.get("/api/articles", (req, res) => {
-    res.json(articles);
-  });
+  app.get('/api/articles', (req, res) => {
+    res.json(articles)
+  })
 
-  app.get("/api/articles/:id", (req, res) => {
-    const article = articles.find(a => a.id.toString() === req.params.id);
-    const index = articles.indexOf(article);
+  app.get('/api/articles/:id', (req, res) => {
+    const article = articles.find(a => a.id.toString() === req.params.id)
+    const index = articles.indexOf(article)
 
-    res.json(articles[index]);
-  });
+    res.json(articles[index])
+  })
 
-  app.patch("/api/articles/:id", (req, res) => {
-    const { body } = req;
-    const article = articles.find(a => a.id.toString() === req.params.id);
-    const index = articles.indexOf(article);
-
-    if (index >= 0) {
-      article.title = body.title;
-      article.content = body.content;
-      articles[index] = article;
-    }
-
-    res.json(article);
-  });
-
-  app.put("/api/articles/:id", (req, res) => {
-    const { body } = req;
-    const article = articles.find(a => a.id.toString() === req.params.id);
-    const index = articles.indexOf(article);
+  app.patch('/api/articles/:id', (req, res) => {
+    const { body } = req
+    const article = articles.find(a => a.id.toString() === req.params.id)
+    const index = articles.indexOf(article)
 
     if (index >= 0) {
-      article.title = body.title;
-      article.content = body.content;
-      articles[index] = article;
+      article.title = body.title
+      article.content = body.content
+      articles[index] = article
     }
 
-    res.json(article);
-  });
+    res.json(article)
+  })
 
-  app.delete("/api/articles/:id", (req, res) => {
-    const article = articles.find(a => a.id.toString() === req.params.id);
-    const index = articles.indexOf(article);
+  app.put('/api/articles/:id', (req, res) => {
+    const { body } = req
+    const article = articles.find(a => a.id.toString() === req.params.id)
+    const index = articles.indexOf(article)
 
-    if (index >= 0) articles.splice(index, 1);
+    if (index >= 0) {
+      article.title = body.title
+      article.content = body.content
+      articles[index] = article
+    }
 
-    res.status(202).send();
-  });
+    res.json(article)
+  })
 
-  app.post("/api/articles", (req, res) => {
+  app.delete('/api/articles/:id', (req, res) => {
+    const article = articles.find(a => a.id.toString() === req.params.id)
+    const index = articles.indexOf(article)
+
+    if (index >= 0) articles.splice(index, 1)
+
+    res.status(202).send()
+  })
+
+  app.post('/api/articles', (req, res) => {
     let id
     if (!articles.length) {
       id = 0
     } else {
-      id = articles[articles.length - 1].id + 1;
+      id = articles[articles.length - 1].id + 1
     }
-    const { body } = req;
+    const { body } = req
 
     const article = {
       id,
       title: body.title,
-      content: body.content
-    };
+      content: body.content,
+    }
 
-    articles.push(article);
+    articles.push(article)
 
-    res.status(201).send(article);
-  });
+    res.status(201).send(article)
+  })
 }

--- a/utils/server-test/services/auth.js
+++ b/utils/server-test/services/auth.js
@@ -1,10 +1,9 @@
 const bcrypt = require('bcrypt')
-const jwt = require('jsonwebtoken');
+const jwt = require('jsonwebtoken')
 
 const secret = 'mySecret'
 
 module.exports = function(app) {
-
   let whitelist = [
     {
       id: 123456,
@@ -19,17 +18,19 @@ module.exports = function(app) {
       password: '$2b$08$ZRrJWppetUmKe5Zlc1Mtj.w51ZTmmx2M4YKUdS4QOBHXq2gzF/zyS', // bcrypt.hashSync('123456')
       permissions: ['admin'],
       token: '',
-    }
+    },
   ]
 
   app.post('/api/auth', (req, res) => {
-    const user = whitelist.find(user => user.email === req.headers.username);
-    if (!user)
-      return res.status(401).send('Invalid user or password');
-    const isPasswordValid = bcrypt.compareSync(req.headers.password, user.password);
+    const user = whitelist.find(user => user.email === req.headers.username)
+    if (!user) return res.status(401).send('Invalid user or password')
+    const isPasswordValid = bcrypt.compareSync(
+      req.headers.password,
+      user.password
+    )
     if (!isPasswordValid)
-      return res.status(401).send('Invalid user or password');
-    const token = jwt.sign({ id: user.id }, secret, { expiresIn: 3600 });
+      return res.status(401).send('Invalid user or password')
+    const token = jwt.sign({ id: user.id }, secret, { expiresIn: 3600 })
     // Saves the token in the whitelist user array
     whitelist = whitelist.map(_user => {
       if (_user.id === user.id) {

--- a/utils/server-test/services/authors.js
+++ b/utils/server-test/services/authors.js
@@ -1,31 +1,40 @@
-const Ipsum = require("bavaria-ipsum");
-const randomDate = require("../utils/random-date-generator")
+const { generateModel } = require('fake-data-generator')
 
-module.exports = function (app) {
-
-  const ipsum = new Ipsum();
-
-  const startDate  = new Date(1970, 1, 1, 0, 0, 0, 0);
-  const finishDate = new Date(1980, 1, 1, 0, 0, 0, 0);
-
-  const formatDate = aIsoDate => {
-    return aIsoDate.toISOString()
+module.exports = function(app) {
+  const model = {
+    config: {
+      locale: 'en',
+    },
+    model: {
+      type: 'Object',
+      value: {
+        id: {
+          type: 'randomNumberBetween',
+          value: [1, 2500000],
+        },
+        name: {
+          type: 'faker',
+          value: 'name.firstName',
+        },
+        lastname: {
+          type: 'faker',
+          value: 'name.lastName',
+        },
+        birthdate: {
+          type: 'faker',
+          value: 'date.between',
+          options: ['1970-01-01', '1996-12-31'],
+        },
+      },
+    },
   }
 
-  const generateRandomAuthor = id => {
-    return {
-      id,
-      name: ipsum.generateWord(),
-      lastname: ipsum.generateWord(),
-      birthdate: formatDate(randomDate(startDate, finishDate))
-    };
-  }
-
-  const authors = [
-    generateRandomAuthor(1),
-    generateRandomAuthor(2),
-    generateRandomAuthor(3)
-  ];
+  const authors = generateModel({
+    amountArg: 40,
+    modelArg: model,
+    inputType: 'object',
+    outputType: 'object',
+  })
 
   app.get("/api/authors", (req, res) => {
     res.json(authors);

--- a/utils/server-test/services/authors.js
+++ b/utils/server-test/services/authors.js
@@ -36,74 +36,74 @@ module.exports = function(app) {
     outputType: 'object',
   })
 
-  app.get("/api/authors", (req, res) => {
-    res.json(authors);
-  });
+  app.get('/api/authors', (req, res) => {
+    res.json(authors)
+  })
 
-  app.get("/api/authors/:id", (req, res) => {
-    const author = authors.find(a => a.id.toString() === req.params.id);
-    const index = authors.indexOf(author);
+  app.get('/api/authors/:id', (req, res) => {
+    const author = authors.find(a => a.id.toString() === req.params.id)
+    const index = authors.indexOf(author)
 
-    res.json(authors[index]);
-  });
+    res.json(authors[index])
+  })
 
-  app.patch("/api/authors/:id", (req, res) => {
-    const { body } = req;
-    const author = authors.find(a => a.id.toString() === req.params.id);
-    const index = authors.indexOf(author);
-
-    if (index >= 0) {
-      author.name = body.name;
-      author.lastname = body.lastname;
-      author.birthdate = body.birthdate;
-      authors[index] = author;
-    }
-
-    res.json(author);
-  });
-
-  app.put("/api/authors/:id", (req, res) => {
-    const { body } = req;
-    const author = authors.find(a => a.id.toString() === req.params.id);
-    const index = authors.indexOf(author);
+  app.patch('/api/authors/:id', (req, res) => {
+    const { body } = req
+    const author = authors.find(a => a.id.toString() === req.params.id)
+    const index = authors.indexOf(author)
 
     if (index >= 0) {
-      author.name = body.name;
-      author.lastname = body.lastname;
-      author.birthdate = body.birthdate;
-      authors[index] = author;
+      author.name = body.name
+      author.lastname = body.lastname
+      author.birthdate = body.birthdate
+      authors[index] = author
     }
 
-    res.json(author);
-  });
+    res.json(author)
+  })
 
-  app.delete("/api/authors/:id", (req, res) => {
-    const author = authors.find(a => a.id.toString() === req.params.id);
-    const index = authors.indexOf(author);
+  app.put('/api/authors/:id', (req, res) => {
+    const { body } = req
+    const author = authors.find(a => a.id.toString() === req.params.id)
+    const index = authors.indexOf(author)
 
-    if (index >= 0) authors.splice(index, 1);
+    if (index >= 0) {
+      author.name = body.name
+      author.lastname = body.lastname
+      author.birthdate = body.birthdate
+      authors[index] = author
+    }
 
-    res.status(202).send();
-  });
+    res.json(author)
+  })
 
-  app.post("/api/authors", (req, res) => {
+  app.delete('/api/authors/:id', (req, res) => {
+    const author = authors.find(a => a.id.toString() === req.params.id)
+    const index = authors.indexOf(author)
+
+    if (index >= 0) authors.splice(index, 1)
+
+    res.status(202).send()
+  })
+
+  app.post('/api/authors', (req, res) => {
     let id
     if (!authors.length) {
       id = 0
     } else {
-      id = authors[authors.length - 1].id + 1;
+      id = authors[authors.length - 1].id + 1
     }
-    const { body } = req;
+    const { body } = req
 
     const author = {
       id,
       name: body.name,
       lastname: body.lastname,
-      birthdate: body.birthdate
-    };
+      birthdate: body.birthdate,
+    }
 
-    authors.push(author);
+    authors.push(author)
 
-    res.status(201).send(author);
-  });
+    res.status(201).send(author)
+  })
 }

--- a/utils/server-test/services/magazines.js
+++ b/utils/server-test/services/magazines.js
@@ -1,8 +1,7 @@
-const Ipsum = require("bavaria-ipsum");
+const Ipsum = require('bavaria-ipsum')
 
-module.exports = function (app) {
-
-  const ipsum = new Ipsum();
+module.exports = function(app) {
+  const ipsum = new Ipsum()
 
   const magazines = [
     {
@@ -10,90 +9,90 @@ module.exports = function (app) {
       name: 'Console log Oriented Programming',
       articles: [1, 2, 3],
       issue: '#20',
-      publisher: ipsum.generateParagraph(1)
+      publisher: ipsum.generateParagraph(1),
     },
     {
       id: 2,
       name: ipsum.generateSentence(),
       articles: [],
       issue: '#13',
-      publisher: ipsum.generateParagraph(1)
+      publisher: ipsum.generateParagraph(1),
     },
     {
       id: 3,
       name: ipsum.generateSentence(),
       articles: [],
       issue: '#7',
-      publisher: ipsum.generateParagraph(1)
-    }
-  ];
+      publisher: ipsum.generateParagraph(1),
+    },
+  ]
 
-  app.get("/api/magazines", (req, res) => {
-    res.json(magazines);
-  });
+  app.get('/api/magazines', (req, res) => {
+    res.json(magazines)
+  })
 
-  app.get("/api/magazines/:id", (req, res) => {
-    const magazine = magazines.find(a => a.id.toString() === req.params.id);
-    const index = magazines.indexOf(magazine);
+  app.get('/api/magazines/:id', (req, res) => {
+    const magazine = magazines.find(a => a.id.toString() === req.params.id)
+    const index = magazines.indexOf(magazine)
 
-    res.json(magazines[index]);
-  });
+    res.json(magazines[index])
+  })
 
-  app.patch("/api/magazines/:id", (req, res) => {
-    const { body } = req;
-    const magazine = magazines.find(a => a.id.toString() === req.params.id);
-    const index = magazines.indexOf(magazine);
-
-    if (index >= 0) {
-      magazine.name = body.name;
-      magazine.editorial = body.editorial;
-      magazine.issue = body.issue;
-      magazine.publisher = body.publisher;
-      magazines[index] = magazine;
-    }
-
-    res.json(magazine);
-  });
-
-  app.put("/api/magazines/:id", (req, res) => {
-    const { body } = req;
-    const magazine = magazines.find(a => a.id.toString() === req.params.id);
-    const index = magazines.indexOf(magazine);
+  app.patch('/api/magazines/:id', (req, res) => {
+    const { body } = req
+    const magazine = magazines.find(a => a.id.toString() === req.params.id)
+    const index = magazines.indexOf(magazine)
 
     if (index >= 0) {
-      magazine.name = body.name;
-      magazine.editorial = body.editorial;
-      magazine.issue = body.issue;
-      magazine.publisher = body.publisher;
-      magazines[index] = magazine;
+      magazine.name = body.name
+      magazine.editorial = body.editorial
+      magazine.issue = body.issue
+      magazine.publisher = body.publisher
+      magazines[index] = magazine
     }
 
-    res.json(magazine);
-  });
+    res.json(magazine)
+  })
 
-  app.delete("/api/magazines/:id", (req, res) => {
-    const magazine = magazines.find(a => a.id.toString() === req.params.id);
-    const index = magazines.indexOf(magazine);
+  app.put('/api/magazines/:id', (req, res) => {
+    const { body } = req
+    const magazine = magazines.find(a => a.id.toString() === req.params.id)
+    const index = magazines.indexOf(magazine)
 
-    if (index >= 0) magazines.splice(index, 1);
+    if (index >= 0) {
+      magazine.name = body.name
+      magazine.editorial = body.editorial
+      magazine.issue = body.issue
+      magazine.publisher = body.publisher
+      magazines[index] = magazine
+    }
 
-    res.status(202).send();
-  });
+    res.json(magazine)
+  })
 
-  app.post("/api/magazines", (req, res) => {
-    const id = magazines[magazines.length - 1].id + 1;
-    const { body } = req;
+  app.delete('/api/magazines/:id', (req, res) => {
+    const magazine = magazines.find(a => a.id.toString() === req.params.id)
+    const index = magazines.indexOf(magazine)
+
+    if (index >= 0) magazines.splice(index, 1)
+
+    res.status(202).send()
+  })
+
+  app.post('/api/magazines', (req, res) => {
+    const id = magazines[magazines.length - 1].id + 1
+    const { body } = req
 
     const magazine = {
       id,
       name: body.name,
       editorial: body.editorial,
       issue: body.issue,
-      publisher: body.publisher
-    };
+      publisher: body.publisher,
+    }
 
-    magazines.push(magazine);
+    magazines.push(magazine)
 
-    res.status(201).send(magazine);
-  });
+    res.status(201).send(magazine)
+  })
 }

--- a/utils/server-test/utils/random-date-generator.js
+++ b/utils/server-test/utils/random-date-generator.js
@@ -1,3 +1,0 @@
-module.exports = (start, end) => {
-    return new Date(start.getTime() + Math.random() * (end.getTime() - start.getTime()));
-}

